addons/todo/models/project_task.py

```diff
@@ -21,6 +21,34 @@ def create(self, vals_list):
                 vals['name'] = _('Untitled to-do')
         return super().create(vals_list)
 
+    def _ensure_onboarding_todo(self):
+        if not self.env.user.has_group('todo.group_onboarding_todo'):
+            self._generate_onboarding_todo(self.env.user)
+            onboarding_group = self.env.ref('todo.group_onboarding_todo').sudo()
+            onboarding_group.users += self.env.user
```

Comment:
I am not sure if that line will trigger the prefetch method. 🤔 

Just to be sure it is not the case we could use write method instead. 🙂 

```suggestion
            onboarding_group.write({'users': [Command.link(self.env.user.id)]})
```

`Command` has to be imported like this:
```py
from odoo import Command
```